marker_rustc_driver/src/conversion/marker.rs

```diff
@@ -304,14 +307,37 @@ impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
 
 impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
     #[must_use]
-    fn to_crate(
-        &self,
-        rustc_crate_id: hir::def_id::CrateNum,
-        rustc_root_mod: &'tcx hir::Mod<'tcx>,
-    ) -> &'ast Crate<'ast> {
-        self.alloc(Crate::new(
-            self.to_crate_id(rustc_crate_id),
-            self.to_items(rustc_root_mod.item_ids),
-        ))
+    fn local_crate(&self) -> &'ast Crate<'ast> {
+        if let Some(krate) = self.krate.get() {
+            return krate;
+        };
+
+        let krate = self.alloc(
+            Crate::builder()
+                .id(self.to_crate_id(hir::def_id::LOCAL_CRATE))
+                .root_mod(self.local_crate_mod())
+                .build(),
+        );
+        self.krate
+            .set(krate)
+            .expect("this can't be filled, since we checked earlier and this stage is single threaded");
```

Comment:
There is [get_or_init](https://doc.rust-lang.org/std/cell/struct.OnceCell.html#method.get_or_init) that allows to avoid unwraps